actionners/kubernetes/drain/drain.go

```diff
@@ -0,0 +1,188 @@
+package drain
+
+import (
+	"context"
+	"fmt"
+	"github.com/falco-talon/falco-talon/internal/events"
+	kubernetes "github.com/falco-talon/falco-talon/internal/kubernetes/client"
+	"github.com/falco-talon/falco-talon/internal/kubernetes/helpers"
+	"github.com/falco-talon/falco-talon/internal/rules"
+	"github.com/falco-talon/falco-talon/utils"
+	"github.com/go-playground/validator/v10"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"regexp"
+	"strings"
+)
+
+const (
+	validatorName = "is_absolut_or_percent"
+	// EvictionKind represents the kind of evictions object
+	EvictionKind = "Eviction"
+	// EvictionSubresource represents the kind of evictions object as pod's subresource
+	EvictionSubresource = "pods/eviction"
+)
+
+type Config struct {
+	MinHealthyReplicas string `mapstructure:"min_healthy_replicas" validate:"omitempty,is_absolut_or_percent"`
+	IgnoreDaemonsets   bool   `mapstructure:"ignore_daemonsets" validate:"omitempty"`
+	IgnoreStatefulSets bool   `mapstructure:"ignore_statefulsets" validate:"omitempty"`
+	GracePeriodSeconds int    `mapstructure:"grace_period_seconds" validate:"omitempty"`
+}
+
+func Action(action *rules.Action, event *events.Event) (utils.LogLine, error) {
+	podName := event.GetPodName()
+	namespace := event.GetNamespaceName()
+	objects := map[string]string{}
+
+	parameters := action.GetParameters()
+	gracePeriodSeconds := new(int64)
+	if val, ok := parameters["grace_period_seconds"].(int); ok {
+		*gracePeriodSeconds = int64(val)
+	}
+
+	client := kubernetes.GetClient()
+	pod, err := client.GetPod(podName, namespace)
+	if err != nil {
+		return utils.LogLine{
+			Objects: objects,
+			Error:   err.Error(),
+			Status:  "failure",
+		}, err
+	}
+
+	node, err := client.GetNodeFromPod(pod)
+	if err != nil {
+		return utils.LogLine{
+			Objects: objects,
+			Error:   err.Error(),
+			Status:  "failure",
+		}, err
+	}
+
+	objects["node"] = node.Name
+
+	pods, err := client.Clientset.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("spec.nodeName=%s", node.GetName()),
+	})
+	if err != nil {
+		return utils.LogLine{
+			Objects: objects,
+			Error:   err.Error(),
+			Status:  "failure",
+		}, err
+	}
+
+	var aggregatedLog utils.LogLine
+	aggregatedLog.Objects = make(map[string]string)
+	var allErrors []string
+	var ignoredPods []string
+
+	for _, p := range pods.Items {
```

Comment:
I guess we could use goroutines with a WaitGroup to send all Evictions request in parallel and accelerate the process

Reply:
Agreed! will work on it